float32[] positions
float32[] velocities
float32[] accelerations
float32[] effort
duration time_from_start